decode.go

```diff
@@ -569,29 +575,41 @@ func (u *unmarshaler) unmarshalList(node *yaml.Node, field protoreflect.FieldDes
 			}
 		default:
 			for _, itemNode := range node.Content {
-				list.Append(u.unmarshalScalar(itemNode, field, false))
+				val, ok := u.unmarshalScalar(itemNode, field, false)
+				if !ok {
+					continue
+				}
+				list.Append(val)
 			}
 		}
 	}
 }
 
 // Unmarshal the map, with explicit handling for maps to messages.
 func (u *unmarshaler) unmarshalMap(node *yaml.Node, field protoreflect.FieldDescriptor, mapVal protoreflect.Map) {
-	if u.checkKind(node, yaml.MappingNode) {
-		mapKeyField := field.MapKey()
-		mapValueField := field.MapValue()
-		for i := 1; i < len(node.Content); i += 2 {
-			keyNode := node.Content[i-1]
-			valueNode := node.Content[i]
-			mapKey := u.unmarshalScalar(keyNode, mapKeyField, true)
-			switch mapValueField.Kind() {
-			case protoreflect.MessageKind, protoreflect.GroupKind:
-				mapValue := mapVal.NewValue()
-				u.unmarshalMessage(valueNode, mapValue.Message().Interface(), false)
-				mapVal.Set(mapKey.MapKey(), mapValue)
-			default:
-				mapVal.Set(mapKey.MapKey(), u.unmarshalScalar(valueNode, mapValueField, false))
+	if !u.checkKind(node, yaml.MappingNode) {
+		return
+	}
```

Comment:
Sorry for the noisy diff here. If you change diff settings in GitHub UI to ignore whitespace diff, it will look better. To be more idiomatic Go, I changed this to early return so I could un-indent the main logic.